sdk/cosmos/azure_data_cosmos/src/routing/global_endpoint_manager.rs

```diff
@@ -0,0 +1,614 @@
+    ///
+    /// # Arguments
+    /// * `endpoint` - The endpoint URL to mark as unavailable for reads
+    pub fn mark_endpoint_unavailable_for_read(&self, endpoint: &str) {
+        self.location_cache
+            .lock()
+            .unwrap()
+            .mark_endpoint_unavailable(endpoint, RequestOperation::Read)
+    }
+
+    /// Marks an endpoint as unavailable for write operations.
+    ///
+    /// # Summary
+    /// Flags the specified endpoint as unavailable for write requests in the location cache.
+    /// This is called by retry policies when write requests fail due to endpoint issues,
+    /// preventing subsequent write operations from being routed to the failing endpoint.
+    /// The endpoint may still be used for read operations if not separately marked unavailable.
+    ///
+    /// # Arguments
+    /// * `endpoint` - The endpoint URL to mark as unavailable for writes
+    pub fn mark_endpoint_unavailable_for_write(&self, endpoint: &str) {
+        self.location_cache
+            .lock()
+            .unwrap()
+            .mark_endpoint_unavailable(endpoint, RequestOperation::Write)
+    }
+
+    /// Determines if a request can utilize multiple write locations.
+    ///
+    /// # Summary
+    /// Evaluates whether the given request can be routed to multiple write regions based
+    /// on the request's operation type and resource type. Returns true only for write
+    /// operations on resources that support multi-master writes (documents and stored
+    /// procedure executions) when the account is configured for multiple write locations.
+    ///
+    /// # Arguments
+    /// * `request` - The Cosmos DB request to evaluate
+    ///
+    /// # Returns
+    /// `true` if the request can use multiple write locations, `false` otherwise
+    pub fn can_use_multiple_write_locations(&self, request: &CosmosRequest) -> bool {
+        !request.is_read_only_request()
+            && self
+                .can_support_multiple_write_locations(request.resource_type, request.operation_type)
+    }
+
+    /// Refreshes account properties and location information from the service.
+    ///
+    /// # Summary
+    /// Fetches the latest Cosmos DB account properties including regional endpoint information
+    /// and updates the location cache. Uses a Moka cache with 600 second TTL to avoid redundant
+    /// service calls. If `force_refresh` is true, invalidates the cache to ensure fresh data.
+    /// The location cache is updated only when new data is fetched (TTL expiry or forced refresh),
+    /// not when serving cached data.
+    ///
+    /// # Arguments
+    /// * `force_refresh` - If true, invalidates cache and forces fresh fetch from service
+    ///
+    /// # Returns
+    /// `Ok(())` if refresh succeeded, `Err` if fetching account properties failed
+    pub async fn refresh_location_async(&self, force_refresh: bool) -> Result<(), Error> {
```

Comment:
No `_async` suffix needed.

Reply:
My recommendation here is to have a `clippy` rule for such cases, so that we can easily avoid such suffix or prefix for any future PR.

Will try to look into this and add a rule if possible.

Reply:
Agreed. I've never created clippy rules before so I don't know how easy it is but it's worth a look!